Analysis/src/EventInfo.cpp

```diff
@@ -271,7 +271,9 @@ LorentzVector EventInfoBase::GetResonanceMomentum(bool useSVfit, bool addMET)
     Lock lock(*mutex);
     if(useSVfit && addMET)
         throw exception("Can't add MET and with SVfit applied.");
-    LorentzVector p4 = GetHiggsTTMomentum(useSVfit) + GetHiggsBB().GetMomentum();
+    LorentzVector p4 (0,0,0,0);
+    if(useSVfit && GetSVFitResults().has_valid_momentum)
+        p4 = GetHiggsTTMomentum(useSVfit) + GetHiggsBB().GetMomentum() ;
```

Comment:
wrong. should also consider cases where useSVfit=false.

Reply:
still wrong.

Reply:
wrong. what if svfit has not converged?

Reply:
There are GetSVFitResults().has_valid_momentum & useSVfit=false. You said that should be calculated also when is false. After, only requiring  valid momentum is wrong, that's what I understood that checked if svfit has not converged, and for that event the p4 remain 0. What is wrong with:

   ```
 if(GetSVFitResults().has_valid_momentum)
        p4 = GetHiggsTTMomentum(useSVfit) + GetHiggsBB().GetMomentum()
```
?
In the case that is not valid it remains p4(0,0,0,0)

Reply:
> What is wrong with:
> ```
> if(GetSVFitResults().has_valid_momentum)
>      p4 = GetHiggsTTMomentum(useSVfit) + GetHiggsBB().GetMomentum()
> ```
> ?

if `useSVfit=false`, `p4` shouldn't depend on whatever SVFit has valid momentum or not.